src/Sylius/Component/Resource/Repository/InMemoryRepository.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Resource\Repository;
+
+use ArrayObject;
+use Pagerfanta\Adapter\ArrayAdapter;
+use Pagerfanta\Pagerfanta;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class InMemoryRepository implements RepositoryInterface
+{
+    /**
+     * @var PropertyAccessor
+     */
+    protected $accessor;
+
+    /**
+     * @var ArrayObject
+     */
+    protected $arrayObject;
+
+    /**
+     * @var string
+     */
+    protected $interface;
+
+    /**
+     * @param string $interface
+     *
+     * @throws \InvalidArgumentException
+     * @throws UnexpectedTypeException
+     */
+    public function __construct($interface)
+    {
+        if (null === $interface) {
+            throw new \InvalidArgumentException('Resource\'s interface needs to be stated.');
+        }
+
+        if (!in_array(ResourceInterface::class, class_implements($interface))) {
+            throw new UnexpectedTypeException($interface, ResourceInterface::class);
+        }
+
+        $this->interface = $interface;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+        $this->arrayObject = new ArrayObject();
+    }
+
+    /**
+     * @param ResourceInterface $resource
+     *
+     * @throws \InvalidArgumentException
+     * @throws UnexpectedTypeException
+     */
+    public function add(ResourceInterface $resource)
+    {
+        if (!in_array($this->interface, class_implements($resource))) {
+            throw new UnexpectedTypeException($resource, $this->interface);
+        }
+
+        if (null === $resource->getId()) {
```

Comment:
Why do we need to have id in order to add resource to database? If we store resources in memory, we can compare them with `===` or using [spl_object_hash](http://php.net/manual/pl/function.spl-object-hash.php).